Ldap authentication

shirlei opened the pull request:
Adds ldap authentication. Uses django-auth-ldap (https://pythonhosted.org/django-auth-ldap/) backend.

benadida commented:
This looks great. Any way to add a handful of tests? I know we are missing a bunch on other auth methods, but better to start now than continue my prior mistake. Let me know if you're up for it. I really want to merge this.

shirlei commented:
Cool... I'll take a look on how to accomplish that, I think it is a good idea!

shirlei commented:
Hi Ben, I've just updated things a bit. I also added a bunch of tests using a online ldap server maintained by forumsys. Please, see what you think about the changes and let me know.
Thank you!
